src/core/useAnimations.tsx

```diff
@@ -37,9 +39,11 @@ export function useAnimations<T extends AnimationClip>(
       })
     )
     return { ref: actualRef, clips, actions, names: clips.map((c) => c.name), mixer }
-  })
+  }
+  const [api, setApi] = React.useState<Api<T>>(makeApi)
   useFrame((state, delta) => mixer.update(delta))
   React.useEffect(() => {
+    setApi(makeApi())
```

Comment:
Perhaps this should be `useMemo` instead. Cleanup can stay in an effect which should address #1594.

Reply:
Agreed, especially because it eliminates the superfluous render caused by `setApi`. I just pushed this change. But how is #1594 addressed as is?

Reply:
I see we'll still need that PR.